### PR TITLE
ctrl: Implement controller remapping

### PR DESCRIPTION
**Pull request by dima-xd**
This is the continiuation of https://github.com/Vita3K/Vita3K/pull/2735 (Thanks to @nn9dev ).

The implementation is not great, so I need opinions what can be done better or what can be also added.

### CODE REVIEW

**Comment by Zangetsu38** [comment]
i can do nothing twith this, you need squash your owner commit in one, for have only 2 coimmit

**Comment by dima-xd** [comment]
> i can do nothing twith this, you need squash your owner commit in one, for have only 2 coimmit

should be good noq

**Comment by Zangetsu38** [comment]
Rebase, fix conflict, and change commit name
Respect format of it

**Comment by dima-xd** [comment]
> Rebase, fix conflict, and change commit name Respect format of it

done

**Comment by Zangetsu38** [comment]
> > Rebase, fix conflict, and change commit name Respect format of it
> 
> done

No, first commit has wrong name
And no need description of preview pr inside it.
And your proper commit name us strange 
Also, build fail here
You need fix fail rebase and fix build

File: vita3k/config/include/config/config.h [hunk 1]
```diff
@@ -148,6 +148,7 @@ enum PerfomanceOverleyPosition {
 // If you are going to implement a dynamic list in the YAML, add it here instead
 // When adding in a new macro for generation, ALL options must be stated.
 #define CONFIG_VECTOR(code)                                                                             \
+    code(std::vector<short>, "controller-binds", std::vector<short>{}, controller_binds)               \
```
Comment (Zangetsu38) [inline vita3k/config/include/config/config.h:151]:
```suggestion
code(std::vector<short>, "controller-binds", std::vector<short>{}, controller_binds)               \
```

File: vita3k/ctrl/include/ctrl/state.h [hunk 2]
```diff
@@ -41,42 +42,42 @@ struct Controller {
...
+    { SDL_CONTROLLER_BUTTON_X, SCE_CTRL_SQUARE },
+    { SDL_CONTROLLER_BUTTON_Y, SCE_CTRL_TRIANGLE },
     { SDL_CONTROLLER_BUTTON_BACK, SCE_CTRL_SELECT },
+    { SDL_CONTROLLER_BUTTON_GUIDE, SCE_CTRL_PSBUTTON },
     { SDL_CONTROLLER_BUTTON_START, SCE_CTRL_START },
     { SDL_CONTROLLER_BUTTON_DPAD_UP, SCE_CTRL_UP },
-    { SDL_CONTROLLER_BUTTON_DPAD_RIGHT, SCE_CTRL_RIGHT },
     { SDL_CONTROLLER_BUTTON_DPAD_DOWN, SCE_CTRL_DOWN },
     { SDL_CONTROLLER_BUTTON_DPAD_LEFT, SCE_CTRL_LEFT },
+    { SDL_CONTROLLER_BUTTON_DPAD_RIGHT, SCE_CTRL_RIGHT },
```
Comment (Zangetsu38) [inline vita3k/ctrl/include/ctrl/state.h:76]: revert both change here, no needed

**Comment by Zangetsu38** [comment]
image
window is too small, set it more bigger

**Comment by dima-xd** [comment]
> image image
> 
> window of rebind is too small, set it more bigger

image
image

now it's bigger

**Comment by Zangetsu38** [comment]
reset size for main window, it is good, only rebind need more big, and reduce size of width for it

**Comment by dima-xd** [comment]
> Contributor

image
made it like this

**Review by Zangetsu38 — APPROVED** [review]

**Pull request merged**